File: boj/b_1935.js
function solution(n, formula, numbers) {
  formula = formula.split("");
  const s = [];
  const table = {};
  const isAlpha = /[A-Z]/;
  formula.forEach((v) => {
    if (isAlpha.test(v) && !table[v]) table[v] = numbers.shift();
  });
  formula.forEach((v, i) => {
    if (isAlpha.test(v)) formula[i] = table[v];
  });
  for (let i = 0; i < formula.length; i += 1) {
    if (!isNaN(formula[i])) s.push(formula[i]);
    else {
      const [a, b] = [s.pop(), s.pop()];
      s.push(eval("" + b + formula[i] + a));
    }
  }
  return s[0].toFixed(2);
}
console.log(solution(5, "ABC*+DE/-", [1, 2, 3, 4, 5]));
console.log(solution(1, "AA+A+", [1]));

File: boj/b_1935.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

n = int(input())
formula = list(input())
s = []
table = {}
for v in formula:
    if v.isalpha() and v not in table:
        table[v] = int(input())
for i, v in enumerate(formula):
    if v.isalpha():
        formula[i] = table[v]
for i in range(len(formula) - 1):
    if type(formula[i]) is int:
        s.append(formula[i])
    else:
        a, b = s.pop(), s.pop()
        s.append(eval(f"{b} {formula[i]} {a}"))
print(f"{s[0]:.2f}")
